pipenv/core.py

```diff
@@ -506,7 +506,7 @@ def ensure_project(
     three=None,
     python=None,
     validate=True,
-    system=False,
+    in_system=False,
     warn=True,
```

Comment:
You should not change the function signature. I only meant the local variable.
Also, I still think the name `system_or_exists` reflects the purpose better.

Reply:
VSCode has followed the local variable and changed it only where it is used. I had to change the one in the function signature as well, otherwise I would have a new one that is not used. Also, the `ensure_project` function is executed 2 times with `system` as a parameter in `core.py`, so I had to change it further down in the code as well.

Please forgive me if I don't quite understand what you mean.

I thought in_system fits better to a boolian, but you are right, yours describes better what is queried.

Reply:
I believe VSCode is doing a disservice ("Bärendienst") here.
The `in_system` variable is defined in L521, and then referenced within the scope of this function in L525 and then L565.
These are the only places that need to be renamed.

Reply:
Then what is the system variable in the function signature for? If I replace system in L521, L525 and L566, the variable in L509 becomes superfluous and is not used.

Reply:
It's used in case the if condition is not met, and it is used by external calling functions. I  would not touch it, because it can create havoc. Only in the scope of the function I would change it to be:
```python
    if project.s.PIPENV_USE_SYSTEM or project.virtualenv_exists:
        system_or_env = True
    else:
        system_or_env = system # defaults to False

    ....
       ensure_pipfile(
        project, validate=validate, skip_requirements=skip_requirements, system=system_or_env
    )
```
```
That keeps the orignal meaning of the variable, but make it clear in the scope of this function and this
fix.

Reply:
`...  
File "/opt/pipenv/pipenv/core.py", line 1861, in do_install
    ensure_project(
  File "/opt/pipenv/pipenv/core.py", line 525, in ensure_project
    if not system_or_exists:
UnboundLocalError: local variable 'system_or_exists' referenced before assignment
`

Unfortunately, that's what I get if I just change what you suggest.

Reply:
Well, it was implicit. I am sorry for the confusion. We both mean to create good software. I know a PR review can be frustrating. Sorry it takes time.